docs/articles/documentation/guides/basic-guides/obtain-client-side-info.md

```diff
@@ -174,6 +174,15 @@ does not finish before the callback, suspend the test until the callback fires.
   >
   > The return value is the only way to obtain data from client functions.
 
+* You cannot use iterable destructuring (e.g., [`spread`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax), [`rest`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Rest_parameters) operators) on non-array objects (e.g. `NodeList`, `HTMLCollection`).
+
+* You cannot use [property shorthands](http://es6-features.org/#PropertyShorthand) to declare the `ClientFunction`'s  `dependencies` option.
+
+* Several [newly added](https://tc39.es/ecma262/) JavaScript features do not operate properly within client functions. These features include:
+    * Array methods [`Array.from()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from), [`Array.of()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/of);  
+    * Keyed collections: [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set), [`WeakMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap), [`WeakSet`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakSet);
+    * Property shorthands for imported values.
```

Comment:
The most straightforward way to fix the tone and decrease the volume is to avoid repeating negative statements each time, like:

----

Client function code does not support the following syntax and capabilities:

* Generators or `async/await` syntax,
* Access to variables defined in the outer scope,

  > You can use arguments to pass data within these functions, except for self-invoking functions that do not accept outside parameters.
   >
   > The return value is the only way to obtain data from client functions.
* Iterable destructuring (e.g., [`spread`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax), [`rest`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Rest_parameters) operators) on non-array objects (e.g. `NodeList`, `HTMLCollection`),
 * [Property shorthands](http://es6-features.org/#PropertyShorthand) in the `ClientFunction`'s  `dependencies` option,
* Array methods [`Array.from()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from), [`Array.of()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/of);
* Keyed collections: [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set), [`WeakMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap), [`WeakSet`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakSet);
* Property shorthands for imported values.

Reply:
And we can do the same for selectors:

https://github.com/DevExpress/testcafe/pull/5357/files#diff-62c91167cb02db75245f03e206ab2a2eR579

Reply:
Does it make sense, what do you think? @DIRECTcut

Reply:
@VasilyStrelyaev makes perfect sence to me. I can see why having negative pattern repeated so much is detrimental. I'll rewrile selectors accordingly